internal/runners/export/runtime.go

```diff
@@ -0,0 +1,106 @@
+package export
+
+import (
+	"strings"
+	"text/template"
+
+	"github.com/ActiveState/cli/internal/analytics"
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/runbits/rationalize"
+	"github.com/ActiveState/cli/internal/runbits/runtime"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/setup"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+type Runtime struct {
+	out       output.Outputer
+	analytics analytics.Dispatcher
+	svcModel  *model.SvcModel
+	auth      *authentication.Auth
+	project   *project.Project
+	cfg       *config.Instance
+}
+
+type RuntimeParams struct {
+	Path string
+}
+
+func NewRuntime(prime primeable) *Runtime {
+	return &Runtime{
+		prime.Output(),
+		prime.Analytics(),
+		prime.SvcModel(),
+		prime.Auth(),
+		prime.Project(),
+		prime.Config(),
+	}
+}
+
+type ErrProjectNotFound struct {
+	Path string
+}
+
+func (e *ErrProjectNotFound) Error() string {
+	return "project not found"
+}
+
+func (e *Runtime) Run(params *RuntimeParams) (rerr error) {
+	defer rationalizeError(&rerr)
+
+	proj := e.project
+	if params.Path != "" {
+		var err error
+		proj, err = project.FromPath(params.Path)
+		if err != nil {
+			return &ErrProjectNotFound{params.Path}
+		}
+	}
+	if proj == nil {
+		return rationalize.ErrNoProject
+	}
+
+	e.out.Notice(locale.Tr("export_runtime_statement", proj.NamespaceString(), proj.Dir()))
+
+	rt, err := runtime.SolveAndUpdate(e.auth, e.out, e.analytics, proj, nil, target.TriggerActivate, e.svcModel, e.cfg, runtime.OptMinimalUI)
```

Comment:
We should probably add a new trigger for this. I don't believe this requires an update to the analytics doc but we should check to make sure.

Reply:
Good catch, thanks!